scripts/check_backend_associated_test_file.py

```diff
@@ -0,0 +1,131 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This script checks if every backend file has an associated test file."""
+
+from __future__ import annotations
+
+import logging
+import os
+import sys
+
+TOPMOST_LEVEL_PATH = './'
+
+# NOTE TO DEVELOPERS: Do not add any new files to this list without asking
+# @U8NWXD first.
+FILES_WITHOUT_ASSOCIATED_TEST_FILES = [
+    # These are data files for testing and have no logic to be tested.
+    'scripts/linters/test_files/invalid_annotations.py',
+    'scripts/linters/test_files/invalid_author.py',
+    'scripts/linters/test_files/invalid_copyright.py',
+    'scripts/linters/test_files/invalid_docstring.py',
+    'scripts/linters/test_files/invalid_import_order.py',
+    'scripts/linters/test_files/invalid_merge_conflict.py',
+    'scripts/linters/test_files/invalid_datastore.py',
+    'scripts/linters/test_files/invalid_no_newline.py',
+    'scripts/linters/test_files/invalid_pycodestyle_error.py',
+    'scripts/linters/test_files/invalid_pylint_id.py',
+    'scripts/linters/test_files/invalid_python_three.py',
+    'scripts/linters/test_files/invalid_request.py',
+    'scripts/linters/test_files/invalid_tabs.py',
+    'scripts/linters/test_files/invalid_todo.py',
+    'scripts/linters/test_files/invalid_urlopen.py',
+    'scripts/linters/test_files/valid.py',
+    'scripts/linters/test_files/valid_py_ignore_pragma.py',
+    'core/tests/build_sources/extensions/CodeRepl.py',
+    'core/tests/build_sources/extensions/DragAndDropSortInput.py',
+    'core/tests/data/failing_tests.py',
+    'core/tests/data/image_constants.py',
+    'core/tests/data/unicode_and_str_handler.py',
+    'proto_files/text_classifier_pb2.py',
+    'proto_files/training_job_response_payload_pb2.py',
+
+    # The files below are to be tested.
+    # GSoC'22: Improve line and branch coverage for the backend and frontend.
+    'scripts/run_custom_eslint_tests.py',
+    'scripts/run_backend_tests.py',
+    'scripts/run_lighthouse_tests.py',
+    'scripts/run_portserver.py',
+    'scripts/run_presubmit_checks.py',
+    'scripts/run_tests.py',
+    'scripts/start.py',
+    'setup.py',
+    'scripts/check_backend_associated_test_file.py',
+    'scripts/third_party_size_check.py',
+    'scripts/linters/warranted_angular_security_bypasses.py',
+    'core/feconf.py',
+    'core/domain/cron_services.py',
+    'core/platform/datastore/cloud_datastore_services.py',
+    'core/platform/transactions/cloud_transaction_services.py',
+]
+
+
+def main() -> None:
+    """Finds the non-empty backend files that lack an associated test file."""
+
+    all_backend_files = []
+    for (root, _, files) in os.walk(TOPMOST_LEVEL_PATH):
```

Comment:
```suggestion
    for root, _, files in os.walk(TOPMOST_LEVEL_PATH):
```
parentheses are unnecessary

Reply:
Removed them now.